### DISCUSSION TITLE
No pre-built standard library found, and ctru-sys not found

### DISCUSSION THREAD

**Discussion by CenTdemeern1** [rust3ds/ctru-rs#182] [Q&A] [answered]
```
No pre-build std found, using build-std
Warning: unable to check if `ctru` debuginfo should be linked: `ctru-sys` not found
```
These two messages appear every time I try to build something. Is there something I can do to mitigate this?
I don't know how I would pre-build std and have it find it, but I have a feeling it might speed up compile times so I'd like to know how.

**Comment by Meziu** [comment, marked as answer]
> These two messages appear every time I try to build something. Is there something I can do to mitigate this?

The "no pre-build" error is meant to appear to notify the user of the use of `build-std` (which at the end of the day, isn't extremely reliable). The other message should not appear if `ctru-sys` was setup correctly, I'll have to see what's the problem with that.

> I don't know how I would pre-build std and have it find it, but I have a feeling it might speed up compile times so I'd like to know how.

You'd need a custom rustup toolchain and build the standard library for the `armv6k-nintendo-3ds` target. It is not a documented process since building `std` requires `ctru-sys` and some other things, which make for a pretty unstable solution.

**Reply by ian-h-chamberlain** [reply]
Just to add to this, if you _did_ want to build your own toolchain, the [`rustc-dev-guide`](https://rustc-dev-guide.rust-lang.org/building/how-to-build-and-run.html) would be the place to start (probably with the `library` profile during `x setup`).

It is kind of a pain though, and `build-std` should still cache well (as far as I know) so there shouldn't normally be much penalty for using it. 

@Meziu I wonder if it would make sense for us to silence that message unless `-v` was passed, since it's expected for most users now they won't have a prebuilt `std`, and I think that message was probably added when it was just a couple of us hacking on `std`.

**Reply by Meziu** [reply]
> @Meziu I wonder if it would make sense for us to silence that message unless `-v` was passed, since it's expected for most users now they won't have a prebuilt `std`, and I think that message was probably added when it was just a couple of us hacking on `std`.

I guess it's fine to suppress it. Using `build-std` is still the preferred method after all.
